docs/usage/aggregated_roles.md

```diff
@@ -0,0 +1,122 @@
+# Aggregated Roles
+
+The current default cluster role for application controller is hard-coded with a specific set of permissions. This cluster role is operator managed and cannot be modified by the customer. In this case administrative user can opt for custom cluster roles and define permissions by creating cluster roles from scratch or go for aggregated cluster role. Using an aggregated cluster role enables users to easily add their own permissions without having to define a completely new cluster role from scratch.
+
+A user can enable creation of aggregated ClusterRole by setting `argocd.spec.aggregatedClusterRoles` field to `true`.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: basic
+spec:
+  aggregatedClusterRoles: true
+```
+
+When `aggregatedClusterRoles` is `true`, the default cluster role for the Argo CD application controller will be created having `aggregationRule` field. This is the base cluster role and a cluster role binding pointing to it will be created and managed by operator.
+
+Example: Configure permissions using aggregated cluster role model for application controller:
+
+**1 ).** Base cluster role
+
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: argocd-argocd-argocd-application-controller
```

Comment:
This is because names of cluster roles are generated using function [GenerateUniqueResourceName](https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/role.go#L50). It has naming convention `<Instance Name>-<Namespace>-<Component Name>` where component name is taken from variables defined [here](https://github.com/argoproj-labs/argocd-operator/blob/master/common/defaults.go#L19), so name generated by function is `argocd-argocd-argocd-application-controller`,